spec/models/manageiq/providers/openstack/cloud_manager/refresher_spec.rb

```diff
@@ -14,92 +14,129 @@
 
     credentials = {:userid => username, :password => password}
     @ems.authentications << FactoryBot.create(:authentication, credentials)
+    @inventory = nil
+  end
+
+  def reset_cache
+    @ems.reset_openstack_handle
+
+    require "fog/openstack"
+    Fog::OpenStack.instance_variable_set(:@version, nil)
+  end
+
+  def refresh_inventory
+    with_vcr do
+      reset_cache
+
+      @ems.refresh
+      @ems.network_manager.refresh
+      @ems.cinder_manager.refresh
+      @ems.swift_manager.refresh
+    end
+
+    @inventory ? assert_inventory_not_changed { @inventory } : @inventory = serialize_inventory
+    @ems.reload
+
+    assert_specific_vm
+    assert_specific_auth_key_pair
+    assert_specific_cloud_volume
+    assert_specific_host_aggregates
+    assert_specific_cloud_network
+    assert_specific_cloud_subnet
+    assert_specific_security_group
+    assert_specific_network_router
   end
 
   it ".ems_type" do
     expect(described_class.ems_type).to eq(:openstack)
   end
 
   it "will perform a full refresh" do
```

Comment:
So I think it'd be good to group all of these in a `context "#refresh"` since the `.ems_type` is testing a completely different method.  We can move the code that sets up the ems with the secrets and the other helper methods into this context since they aren't needed by other tests.

Reply:
Would all the refresh tests go in `context "#refresh"` as well?

Reply:
Yes